### PR TITLE
evaluateRulesetsCommand improvements

### PR DESCRIPTION
**Pull request by jasonwaters**
## Description

This PR improves the `evaluateRulesets` and `subscribeRulesetItems` commands for consistency.

### evaluateRulesets
evaluateRulesets now has the same api as `sendEvent`.

Before the change:
```js
window.alloy("evaluateRulesets", {
  action: "share-social-media",
});
```

After the change:
```js
window.alloy("evaluateRulesets", {
  renderDecisions: true,
  decisionContext: {
    action: "share-social-media",
  },
});
```
This ☝️  is consistent with `sendEvent` and allows users to specify whether to `renderDecisions` or not.

### subscribeRulesetItems
subscribeRulesetItems now returns an array of propositions the same way `sendEvent` does.

Before the change:
```js
window.alloy("subscribeRulesetItems", {
  surface: "mobileapp://com.adobe.aguaAppIos",
  schemas: ["https://ns.adobe.com/personalization/message/feed-item"],
  callback: (result) => {
    console.log("subscribeRulesetItems", result);
  },
});

///the result object sent to the callback
const result = {
  items: [
    {
      schema: "https://ns.adobe.com/personalization/message/feed-item",
      data: {
        expiryDate: 1712190456,
        publishedDate: 1678098240000,
        qualifiedDate: 1692378210224,
        displayedDate: 1692378210230,
        meta: {
          feedName: "Winter Promo",
          surface: "mobileapp://com.adobe.sampleApp/feed/promos",
        },
        content: {
          imageUrl: "img/twitter.png",
          actionTitle: "Shop the sale!",
          actionUrl: "https://luma.com/sale",
          body: "Posting on social media helps us spread the word.",
          title: "Thanks for sharing!",
        },
        contentType: "application/json",
      },
      id: "cfcb1af7-7bc2-45b2-a86a-0aa93fe69ce7",
    },
    {
      schema: "https://ns.adobe.com/personalization/message/feed-item",
      data: {
        expiryDate: 1712190456,
        publishedDate: 1678184640000,
        qualifiedDate: 1692378210224,
        displayedDate: 1692378210230,
        meta: {
          feedName: "Winter Promo",
          surface: "mobileapp://com.adobe.sampleApp/feed/promos",
        },
        content: {
          imageUrl: "img/gold-coin.jpg",
          actionTitle: "Shop the sale!",
          actionUrl: "https://luma.com/sale",
          body: "Now you're ready to earn!",
          title: "Funds deposited!",
        },
        contentType: "application/json",
      },
      id: "0263e171-fa32-4c7a-9611-36b28137a81d",
    },
  ],
};



```

After the change:
```js
window.alloy("subscribeRulesetItems", {
  surfaces: ["mobileapp://com.adobe.aguaAppIos"],
  schemas: ["https://ns.adobe.com/personalization/message/feed-item"],
  callback: (result) => {
    console.log("subscribeRulesetItems", result);
  },
});


///the result object sent to the callback
const result = {
  propositions: [
    {
      scopeDetails: {
        decisionProvider: "AJO",
        characteristics: {
          eventToken:
            "eyJtZXNzYWdlRXhlY3V0aW9uIjp7Im1lc3NhZ2VFeGVjdXRpb25JRCI6Ik5BIiwibWVzc2FnZUlEIjoiMDJjNzdlYTgtN2MwZS00ZDMzLTgwOTAtNGE1YmZkM2Q3NTAzIiwibWVzc2FnZVR5cGUiOiJtYXJrZXRpbmciLCJjYW1wYWlnbklEIjoiMzlhZThkNGItYjU1ZS00M2RjLWExNDMtNzdmNTAxOTViNDg3IiwiY2FtcGFpZ25WZXJzaW9uSUQiOiJiZDg1ZDllOC0yMDM3LTQyMmYtYjZkMi0zOTU3YzkwNTU5ZDMiLCJjYW1wYWlnbkFjdGlvbklEIjoiYjQ3ZmRlOGItNTdjMS00YmJlLWFlMjItNjRkNWI3ODJkMTgzIiwibWVzc2FnZVB1YmxpY2F0aW9uSUQiOiJhZTUyY2VkOC0yMDBjLTQ5N2UtODc4Ny1lZjljZmMxNzgyMTUifSwibWVzc2FnZVByb2ZpbGUiOnsiY2hhbm5lbCI6eyJfaWQiOiJodHRwczovL25zLmFkb2JlLmNvbS94ZG0vY2hhbm5lbHMvd2ViIiwiX3R5cGUiOiJodHRwczovL25zLmFkb2JlLmNvbS94ZG0vY2hhbm5lbC10eXBlcy93ZWIifSwibWVzc2FnZVByb2ZpbGVJRCI6ImY1Y2Q5OTk1LTZiNDQtNDIyMS05YWI3LTViNTMzOGQ1ZjE5MyJ9fQ==",
        },
        strategies: [
          {
            strategyID: "3VQe3oIqiYq2RAsYzmDTSf",
            treatmentID: "yu7rkogezumca7i0i44v",
          },
        ],
        activity: {
          id: "39ae8d4b-b55e-43dc-a143-77f50195b487#b47fde8b-57c1-4bbe-ae22-64d5b782d183",
        },
        correlationID: "02c77ea8-7c0e-4d33-8090-4a5bfd3d7503",
      },
      id: "1a3d874f-39ee-4310-bfa9-6559a10041a4",
      items: [
        {
          schema: "https://ns.adobe.com/personalization/message/feed-item",
          data: {
            expiryDate: 1712190456,
            publishedDate: 1677752640000,
            meta: {
              feedName: "Winter Promo",
              surface: "mobileapp://com.adobe.sampleApp/feed/promos",
            },
            content: {
              imageUrl: "https://target.jasonwaters.dev/img/lumon.png",
              actionTitle: "Shop the sale!",
              actionUrl: "https://luma.com/sale",
              body: "a handshake is available upon request.",
              title: "Welcome to Lumon!",
            },
            contentType: "application/json",
            qualifiedDate: 1694450969990,
            displayedDate: 1694450969993,
          },
          id: "a48ca420-faea-467e-989a-5d179d9f562d",
        },
        {
          schema: "https://ns.adobe.com/personalization/message/feed-item",
          data: {
            expiryDate: 1712190456,
            publishedDate: 1677839040000,
            meta: {
              feedName: "Winter Promo",
              surface: "mobileapp://com.adobe.sampleApp/feed/promos",
            },
            content: {
              imageUrl:
                "https://media.giphy.com/media/l0Ex3vQtX5VX2YtAQ/giphy.gif",
              actionTitle: "Shop the sale!",
              actionUrl: "https://luma.com/sale",
              body: "Great job, you completed your profile.",
              title: "Achievement Unlocked!",
            },
            contentType: "application/json",
            qualifiedDate: 1694450969990,
            displayedDate: 1694450969993,
          },
          id: "b7173290-588f-40c6-a05c-43ed5ec08b28",
        },
      ],
      scope: "web://target.jasonwaters.dev/aep.html",
    },
    {
      scopeDetails: {
        decisionProvider: "AJO",
        characteristics: {
          eventToken:
            "eyJtZXNzYWdlRXhlY3V0aW9uIjp7Im1lc3NhZ2VFeGVjdXRpb25JRCI6Ik5BIiwibWVzc2FnZUlEIjoiMDJjNzdlYTgtN2MwZS00ZDMzLTgwOTAtNGE1YmZkM2Q3NTAzIiwibWVzc2FnZVR5cGUiOiJtYXJrZXRpbmciLCJjYW1wYWlnbklEIjoiMzlhZThkNGItYjU1ZS00M2RjLWExNDMtNzdmNTAxOTViNDg3IiwiY2FtcGFpZ25WZXJzaW9uSUQiOiJiZDg1ZDllOC0yMDM3LTQyMmYtYjZkMi0zOTU3YzkwNTU5ZDMiLCJjYW1wYWlnbkFjdGlvbklEIjoiYjQ3ZmRlOGItNTdjMS00YmJlLWFlMjItNjRkNWI3ODJkMTgzIiwibWVzc2FnZVB1YmxpY2F0aW9uSUQiOiJhZTUyY2VkOC0yMDBjLTQ5N2UtODc4Ny1lZjljZmMxNzgyMTUifSwibWVzc2FnZVByb2ZpbGUiOnsiY2hhbm5lbCI6eyJfaWQiOiJodHRwczovL25zLmFkb2JlLmNvbS94ZG0vY2hhbm5lbHMvd2ViIiwiX3R5cGUiOiJodHRwczovL25zLmFkb2JlLmNvbS94ZG0vY2hhbm5lbC10eXBlcy93ZWIifSwibWVzc2FnZVByb2ZpbGVJRCI6ImY1Y2Q5OTk1LTZiNDQtNDIyMS05YWI3LTViNTMzOGQ1ZjE5MyJ9fQ==",
        },
        strategies: [
          {
            strategyID: "3VQe3oIqiYq2RAsYzmDTSf",
            treatmentID: "yu7rkogezumca7i0i44v",
          },
        ],
        activity: {
          id: "39ae8d4b-b55e-43dc-a143-77f50195b487#b47fde8b-57c1-4bbe-ae22-64d5b782d183",
        },
        correlationID: "02c77ea8-7c0e-4d33-8090-4a5bfd3d7503",
      },
      id: "1ae11bc5-96dc-41c7-8f71-157c57a5290e",
      items: [
        {
          schema: "https://ns.adobe.com/personalization/message/feed-item",
          data: {
            expiryDate: 1712190456,
            publishedDate: 1678098240000,
            meta: {
              feedName: "Winter Promo",
              surface: "mobileapp://com.adobe.sampleApp/feed/promos",
            },
            content: {
              imageUrl: "https://target.jasonwaters.dev/img/twitter.png",
              actionTitle: "Shop the sale!",
              actionUrl: "https://luma.com/sale",
              body: "Posting on social media helps us spread the word.",
              title: "Thanks for sharing!",
            },
            contentType: "application/json",
            qualifiedDate: 1694450973121,
            displayedDate: 1694450973128,
          },
          id: "cfcb1af7-7bc2-45b2-a86a-0aa93fe69ce7",
        },
      ],
      scope: "web://target.jasonwaters.dev/aep.html",
    },
    {
      scopeDetails: {
        decisionProvider: "AJO",
        characteristics: {
          eventToken:
            "eyJtZXNzYWdlRXhlY3V0aW9uIjp7Im1lc3NhZ2VFeGVjdXRpb25JRCI6Ik5BIiwibWVzc2FnZUlEIjoiMDJjNzdlYTgtN2MwZS00ZDMzLTgwOTAtNGE1YmZkM2Q3NTAzIiwibWVzc2FnZVR5cGUiOiJtYXJrZXRpbmciLCJjYW1wYWlnbklEIjoiMzlhZThkNGItYjU1ZS00M2RjLWExNDMtNzdmNTAxOTViNDg3IiwiY2FtcGFpZ25WZXJzaW9uSUQiOiJiZDg1ZDllOC0yMDM3LTQyMmYtYjZkMi0zOTU3YzkwNTU5ZDMiLCJjYW1wYWlnbkFjdGlvbklEIjoiYjQ3ZmRlOGItNTdjMS00YmJlLWFlMjItNjRkNWI3ODJkMTgzIiwibWVzc2FnZVB1YmxpY2F0aW9uSUQiOiJhZTUyY2VkOC0yMDBjLTQ5N2UtODc4Ny1lZjljZmMxNzgyMTUifSwibWVzc2FnZVByb2ZpbGUiOnsiY2hhbm5lbCI6eyJfaWQiOiJodHRwczovL25zLmFkb2JlLmNvbS94ZG0vY2hhbm5lbHMvd2ViIiwiX3R5cGUiOiJodHRwczovL25zLmFkb2JlLmNvbS94ZG0vY2hhbm5lbC10eXBlcy93ZWIifSwibWVzc2FnZVByb2ZpbGVJRCI6ImY1Y2Q5OTk1LTZiNDQtNDIyMS05YWI3LTViNTMzOGQ1ZjE5MyJ9fQ==",
        },
        strategies: [
          {
            strategyID: "3VQe3oIqiYq2RAsYzmDTSf",
            treatmentID: "yu7rkogezumca7i0i44v",
          },
        ],
        activity: {
          id: "39ae8d4b-b55e-43dc-a143-77f50195b487#b47fde8b-57c1-4bbe-ae22-64d5b782d183",
        },
        correlationID: "02c77ea8-7c0e-4d33-8090-4a5bfd3d7503",
      },
      id: "d1f7d411-a549-47bc-a4d8-c8e638b0a46b",
      items: [
        {
          schema: "https://ns.adobe.com/personalization/message/feed-item",
          data: {
            expiryDate: 1712190456,
            publishedDate: 1678184640000,
            meta: {
              feedName: "Winter Promo",
              surface: "mobileapp://com.adobe.sampleApp/feed/promos",
            },
            content: {
              imageUrl: "https://media.giphy.com/media/ADgfsbHcS62Jy/giphy.gif",
              actionTitle: "Shop the sale!",
              actionUrl: "https://luma.com/sale",
              body: "Now you're ready to earn!",
              title: "Funds deposited!",
            },
            contentType: "application/json",
            qualifiedDate: 1694450971742,
            displayedDate: 1694450971748,
          },
          id: "0263e171-fa32-4c7a-9611-36b28137a81d",
        },
      ],
      scope: "web://target.jasonwaters.dev/aep.html",
    },
  ],
};
```

This ☝️ change makes the result object provided to the callback consistent with what is returned by `alloy('sendEvent').then(result => {})`.  It also allows users to specify any combination of surfaces and/or schemas to filter the result by.  If no surfaces or schemas are provided, all surfaces and schemas will be returned.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/components/DecisioningEngine/createSubscribeRulesetItems.js [hunk 1]
```diff
@@ -28,41 +28,44 @@ const validateOptions = ({ options }) => {
...
+      propositions: propositions
+        .filter(payload =>
+          surfacesFilter ? surfacesFilter.includes(payload.scope) : true
+        )
+        .map(payload => {
+          const { items = [] } = payload;
+          return {
+            ...payload,
+            items: items.filter(item =>
+              schemasFilter ? schemasFilter.includes(item.schema) : true
```
Comment (shandilya3) [inline src/components/DecisioningEngine/createSubscribeRulesetItems.js:57]: Shall we add check for invalid item ? 
`
if (!item || !item.schema) {
            return false; 
          }`

Comment (shandilya3) [reply src/components/DecisioningEngine/createSubscribeRulesetItems.js:57]: Similar validation for payload above
```
 if (!payload || !payload.scope) {
        return false; 
      }
```

**Review by shandilya3 — APPROVED** [review]
Overall LGTM

File: src/components/DecisioningEngine/createSubscribeRulesetItems.js (hunk 1 above)
Comment (jasonwaters) [reply src/components/DecisioningEngine/createSubscribeRulesetItems.js:57]: @shandilya3 not clear on what you are suggesting here and what it accomplishes.  Can you share more?

Comment (jasonwaters) [reply src/components/DecisioningEngine/createSubscribeRulesetItems.js:57]: item will never be undefined, nor will payload.

Comment (shandilya3) [reply src/components/DecisioningEngine/createSubscribeRulesetItems.js:57]: In that case the check is not needed.

**Pull request merged**